opal-gwt-client/src/main/java/org/obiba/opal/web/gwt/app/client/project/genotypes/ProjectGenotypesPresenter.java

```diff
@@ -0,0 +1,250 @@
+        uploadVcfFile(event.getFile(), event.getName());
+      }
+    });
+
+    addRegisteredHandler(GenotypesMappingEditRequestEvent.getType(), new GenotypesMappingEditRequestEvent.GenotypesMappingEditRequestHandler() {
+      @Override
+      public void onGenotypesMappingEditRequest(GenotypesMappingEditRequestEvent event) {
+        updateGenotypesMapping(event.getGenotypesMapping());
+      }
+    });
+  }
+
+  @Override
+  protected void onBind() {
+    super.onBind();
+  }
+
+  public void initialize(ProjectDto dto) {
+    projectDto = dto;
+    refresh();
+  }
+
+  @Override
+  public void onDownloadVcfFiles() {
+  }
+
+  @Override
+  public void onImportVcfFiles() {
+    vcfFileUploadModalPresenterModalProvider.get();
+  }
+
+  @Override
+  public void onEditMappingTable() {
+    ProjectGenotypeEditMappingTableModalPresenter presenter = projectGenotypeEditMappingTableModalPresenterModalProvider.create();
+    presenter.setMappingTables(mappingTables);
+    presenter.setGenotypesMapping(currentGenotypesMapping());
+    projectGenotypeEditMappingTableModalPresenterModalProvider.show();
+  }
+
+  @Override
+  public void onRemoveVcfFile(VCFSummaryDto vcfSummaryDto) {
+    removeVcfFile(vcfSummaryDto.getName());
+  }
+
+  @Override
+  public void onDownloadVcfFile(VCFSummaryDto vcfSummaryDto) {
+
+  }
+
+  @Override
+  public void onDownloadStatistics(VCFSummaryDto vcfSummaryDto) {
+
+  }
+
+  public void refresh() {
+    getGenotypesSummary();
+    getMappingTable();
+    getVcfTables();
+  }
+
+  public void getGenotypesSummary() {
+    ResourceRequestBuilderFactory.<VCFSamplesSummaryDto>newBuilder()
+      .forResource(UriBuilders.PROJECT_GENOTYPES_SUMMARY.create().build(projectDto.getName()))
+      .withCallback(new ResourceCallback<VCFSamplesSummaryDto>() {
+        @Override
+        public void onResource(Response response, VCFSamplesSummaryDto summary) {
+          logger.info("Received Genotypes summary");
+          getView().setGenotypesSummary(summary);
+        }
+      }).get().send();
+  }
+
+  private void getVcfTables() {
+    getView().beforeRenderRows();
+    ResourceRequestBuilderFactory.<JsArray<VCFSummaryDto>>newBuilder()
+      .forResource(UriBuilders.PROJECT_VCF_STORE_VCFS.create().build(projectDto.getName()))
+      .withCallback(new ResourceCallback<JsArray<VCFSummaryDto>>() {
+        @Override
+        public void onResource(Response response, JsArray<VCFSummaryDto> summaries) {
+          getView().renderRows(summaries);
+          getView().afterRenderRows();
+        }
+      })
+      .get()
+      .send();
+  }
+
+  private void getMappingTable() {
+    getView().setGenotypesMapping(currentGenotypesMapping());
+    Map<String, String> params = Maps.newHashMap();
+    params.put(ENTITY_TYPE_PARAM, ENTITY_TYPE);
+
+    ResourceRequestBuilderFactory.<JsArray<TableDto>>newBuilder()
+        .forResource(UriBuilders.DATASOURCE_TABLES.create().query(params).build(projectDto.getName()))
```

Comment:
The list of tables should not be restricted to the ones of the project's datasource. Instead of that you should get all tables with "Sample" entity type.

GET /datasources/tables?entityType=Sample